core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -202,12 +223,33 @@ public void registerGlobalSystemConnector(GlobalSystemConnector connector)
             }
 
             CatalogConnector catalog = catalogFactory.createCatalog(GlobalSystemConnector.CATALOG_HANDLE, GlobalSystemConnector.NAME, connector);
-            if (catalogs.putIfAbsent(GlobalSystemConnector.NAME, catalog) != null) {
+            if (activeCatalogs.putIfAbsent(GlobalSystemConnector.NAME, catalog) != null) {
                 throw new IllegalStateException("Global system catalog already registered");
             }
+            allCatalogs.put(GlobalSystemConnector.CATALOG_HANDLE, catalog);
         }
         finally {
             catalogsUpdateLock.unlock();
         }
     }
+
+    static CatalogVersion computeCatalogVersion(String catalogName, String connectorName, Map<String, String> properties)
```

Comment:
Didn't we say something about catalog versions being lexical sortable? Hashing doesn't seem to satisfy that. If it's not a real constraint, maybe just remove the comment about it being lexical sortable since that would be an implementation detail from the connector side.